### ISSUE TITLE
EKS Terraform destroy may fail and leave lingering resources

### ISSUE THREAD

**Issue by adrienthebo** [gitpod-io/gitpod#11092]
### Bug description

When Terraform is tearing down EKS resources, issues such as order dependent failures or transient issues with AWS may prevent resources from being destroyed. This will leave lingering resources in our AWS infrastructure that could potentially incur billing charges.

[Case in point](https://werft.gitpod-dev.com/job/gitpod-custom-nvn-fix-int-test.3):

```
│ Error: error deleting EC2 Subnet (subnet-0d50704431c2cecef): DependencyViolation: The subnet 'subnet-0d50704431c2cecef' has dependencies and cannot be deleted.
│ 	status code: 400, request id: 8edd7395-d804-4330-afb5-96ac7643d261
│
│
╵
╷
│ Error: error detaching EC2 Internet Gateway (igw-08af7b1bea577cf49) from VPC (vpc-0c66285046f7f0e67): DependencyViolation: Network vpc-0c66285046f7f0e67 has some mapped public address(es). Please unmap those public address(es) before detaching the gateway.
│ 	status code: 400, request id: 63833b38-ad37-4f5a-a661-cb7a787b515f
│
│
╵
╷
│ Error: error deleting EC2 Subnet (subnet-0215aeff4a391867a): DependencyViolation: The subnet 'subnet-0215aeff4a391867a' has dependencies and cannot be deleted.
│ 	status code: 400, request id: 9a629494-eeaa-49e9-9d3a-cae0994e3635
│
│
```

### Steps to reproduce

Run the EKS tests wait for a faulty Terraform plan that has incorrect dependencies.

### Resolution

There are a handful of things we can do.

1. Label AWS resources with the terraform test ID so that we can look for stale or orphaned resources.
2. Label AWS resources with a "destroy-after" date so we can automatically expire stuck resources.
3. Run `terraform destroy ... || terraform destroy`; if the first destroy doesn't work then we can keep whacking AWS with the delete stick until we get a resource application order that works.

**Comment by adrienthebo** [comment]
Noting this for later, it looks like we're leaving linger ELBs. I _suspect_ that ELBs are being created by EKS, but when we delete the EKS cluster via terraform the ELBs don't get wiped. This is of specific note:

```
            },
            "Attachment": {
                "AttachTime": "2022-07-01T18:09:41.000Z",
                "AttachmentId": "eni-attach-044a679e397457c77",
                "DeleteOnTermination": false,
                "DeviceIndex": 1,
                "NetworkCardIndex": 0,
                "InstanceOwnerId": "amazon-elb",
                "Status": "attached"
            },
--
                "PublicIp": "34.253.153.148"
            },
            "Attachment": {
                "AttachTime": "2022-07-01T18:09:36.000Z",
                "AttachmentId": "eni-attach-09a71ddca4caf9b05",
                "DeleteOnTermination": false,
                "DeviceIndex": 1,
                "NetworkCardIndex": 0,
                "InstanceOwnerId": "amazon-elb",
                "Status": "attached"
            },
```

**EDIT**: Yep, we have stale ELBs which are attached to the subnets, those need to be destroyed before we can fully tear down the subnets and VPC. Will track that in a separate ticket.

**Comment by adrienthebo** [comment]
We also may have lingering security groups that are preventing the deletion; we'll have to do more investigation.

**Comment by nandajavarma** [comment]
Fixed by #11093
